include/alpaka/vec/Vec.hpp

```diff
@@ -275,8 +276,9 @@ namespace alpaka
         ALPAKA_FN_HOST_ACC friend constexpr auto operator+(Vec const& p, Vec const& q) -> Vec
         {
             Vec r;
-            for(typename TDim::value_type i = 0; i < TDim::value; ++i)
-                r[i] = p[i] + q[i];
+            if constexpr(TDim::value > 0)
+                for(typename TDim::value_type i = 0; i < TDim::value; ++i)
+                    r[i] = p[i] + q[i];
```

Comment:
```suggestion
            if constexpr(TDim::value > 0)
            {
                for(typename TDim::value_type i = 0; i < TDim::value; ++i)
                    r[i] = p[i] + q[i];
            }
```

Also for the other instances.